behaviour/graql/language/insert.feature

```diff
@@ -200,64 +295,1159 @@ Feature: Graql Insert Query
+      |      | check | value |
+      | REF0 | key   | ref:0 |
+      | REF1 | key   | ref:1 |
+      | REF2 | key   | ref:2 |
+    Then uniquely identify answer concepts
+      | p    | c    | r    |
+      | REF0 | REF2 | REF1 |
+
+
+  Scenario: an additional role player can be inserted into every relation matching a pattern
+    Given graql insert
+      """
+      insert
+      $p isa person, has name "Ruth", has ref 0;
+      $r (employee: $p) isa employment, has ref 1;
+      $s (employee: $p) isa employment, has ref 2;
+      $c isa company, has name "The Boring Company", has ref 3;
+      """
+    Given the integrity is validated
+    When graql insert
+      """
+      match
+        $r isa employment, has ref $ref;
+        $c isa company;
+      insert
+        $r (employer: $c) isa employment;
+      """
+    When the integrity is validated
+    Then get answers of graql query
+      """
+      match $r (employer: $c, employee: $p) isa employment; get;
+      """
+    Then concept identifiers are
+      |      | check | value |
+      | RUTH | key   | ref:0 |
+      | EMP1 | key   | ref:1 |
+      | EMP2 | key   | ref:2 |
+      | COMP | key   | ref:3 |
+    Then uniquely identify answer concepts
+      | p    | c    | r    |
+      | RUTH | COMP | EMP1 |
+      | RUTH | COMP | EMP2 |
+
+
+  Scenario: an additional duplicate role player can be inserted into a relation
+    Given graql insert
+      """
+      insert $p isa person, has ref 0; $r (employee: $p) isa employment, has ref 1;
+      """
+    Given the integrity is validated
+    When graql insert
+      """
+      match $r isa employment; $p isa person; insert $r (employee: $p) isa employment;
+      """
+    When the integrity is validated
+    Then get answers of graql query
+      """
+      match $r (employee: $p, employee: $p) isa employment; get;
+      """
+    Then concept identifiers are
+      |      | check | value |
+      | REF0 | key   | ref:0 |
+      | REF1 | key   | ref:1 |
+    Then uniquely identify answer concepts
+      | p    | r    |
+      | REF0 | REF1 |
+
+
+  Scenario: inserting a roleplayer in a relation that can't have that roleplayer throws on commit
+    When graql define
+      """
+      define
+      tennis-group sub relation, relates tennis-player;
+      person plays tennis-player;
+      """
+    When the integrity is validated
+    Then graql insert throws
+      """
+      insert
+      $r (tennis-player: $p) isa employment, has ref 0;
+      $p isa person, has ref 1;
+      """
+    Then the integrity is validated
+
+
+  Scenario: inserting a roleplayer that can't play the role throws on commit
+    Then graql insert throws
+      """
+      insert
+      $r (employer: $p) isa employment, has ref 0;
+      $p isa person, has ref 1;
+      """
+    Then the integrity is validated
+
+
+  Scenario: inserting a roleplayer that can't play the role, yet is the parent of a type that can play the role, is still invalid and throws on commit
```

Comment:
shorten

Reply:
`inserting a roleplayer that is only the parent of a valid type is invalid and throws on commit`